rocAL/source/decoders/video/ffmpeg_video_decoder.cpp

```diff
@@ -138,22 +141,26 @@ VideoDecoder::Status FFmpegVideoDecoder::Initialize(const char *src_filename, in
     _src_filename = src_filename;
     if (avformat_open_input(&_fmt_ctx, src_filename, NULL, NULL) < 0) {
         ERR("Couldn't Open video file " + STR(src_filename));
+        Release();
         return Status::FAILED;
     }
     if (avformat_find_stream_info(_fmt_ctx, NULL) < 0) {
         ERR("av_find_stream_info error");
+        Release();
         return Status::FAILED;
     }
     ret = av_find_best_stream(_fmt_ctx, AVMEDIA_TYPE_VIDEO, -1, -1, NULL, 0);
     if (ret < 0) {
         ERR("Could not find %s stream in input file " +
```

Comment:
Can we generalize this error handling since all of them need only AVMEDIA_TYPE_VIDEO?